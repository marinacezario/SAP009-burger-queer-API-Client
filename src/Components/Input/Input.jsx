import React from "react";

export function Input({ type, value, onChange, placeholder, className }) {
  return (
    <input
      type={type}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      className={className}
    />
  );
}
