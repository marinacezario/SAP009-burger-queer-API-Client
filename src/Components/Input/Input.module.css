.inputs {
    border: none;
    border-bottom: solid .5rem white;
    background: #4B4447;
    height: 5rem;
    margin-bottom: 10vh;
    font-size: 2.6rem;
    color: white;
    width: 80%;
}

.inputs::placeholder {
    font-size: 2.6rem;
    color:white;
}
