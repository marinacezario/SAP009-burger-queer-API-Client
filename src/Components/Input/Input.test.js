import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Input } from "./Input";
import "@testing-library/jest-dom";

describe("Input component", () => {
  const mockOnChange = jest.fn();
  test("renders without crashing", () => {
    render(<Input type="email" placeholder="Email" onChange={mockOnChange} />);

    const email_input = screen.getByPlaceholderText("Email");

    expect(email_input).toBeInTheDocument();
    expect(email_input).toMatchSnapshot();
    expect(email_input).toHaveAttribute("type", "email");
 
  });
  test("calls onChange", () => {
    render(<Input type="email" placeholder="Email" onChange={mockOnChange} />);

    const email_input = screen.getByPlaceholderText("Email");

    const newValue = "example@example.com";

    fireEvent.change(email_input, { target: { value: newValue } });

    expect(mockOnChange).toHaveBeenCalled();
  });
});
