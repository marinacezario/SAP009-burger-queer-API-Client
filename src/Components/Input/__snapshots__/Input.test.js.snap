// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders without crashing 1`] = `
<input
  placeholder="Email"
  type="email"
  value=""
/>
`;
